airbyte-integrations/connectors/destination-clickhouse/src/main/kotlin/io/airbyte/integrations/destination/clickhouse/write/transform/ClickhouseCoercer.kt

```diff
@@ -69,12 +70,14 @@ class ClickhouseCoercer : ValueCoercer {
 
     override fun validate(value: EnrichedAirbyteValue): EnrichedAirbyteValue {
         when (val abValue = value.abValue) {
-            is NumberValue ->
-                if (abValue.value <= DECIMAL128_MIN || abValue.value >= DECIMAL128_MAX) {
+            is NumberValue -> {
+                val scaled = abValue.value.multiply(DECIMAL_SCALE_MULTIPLIER)
+                if (scaled <= DECIMAL128_MIN || scaled >= DECIMAL128_MAX) {
```

Comment:
nit: I feel like it would be clearer to scale the min value rather than the input value
```kotlin
// precision=38, scale=9
if (value < DECIMAL128_MIN.scaleByPowerOfTen(-9) || ...) {
```

Reply:
I was just copying their code, but this might be a better approach. Let me try it.